website/blog/2023-03-17-databend-weekly-85.md

```diff
@@ -0,0 +1,186 @@
+
+## Code Corner
+
+Discover some fascinating code snippets or projects that showcase our work or learning journey.
+
+### Run Multiple Futures Parallel
+
+Are you interested in how to run futures in parallel? It is worth mentioning that Databend has greatly improved the scanning performance in situations with a huge number of files by utilizing this technique.
+
+The following code, which is less than 30 lines long, will introduce you to how it all works.
+
+```rust
+/// Run multiple futures parallel
+/// using a semaphore to limit the parallelism number, and a specified thread pool to run the futures.
+/// It waits for all futures to complete and returns their results.
+pub async fn execute_futures_in_parallel<Fut>(
+    futures: impl IntoIterator<Item = Fut>,
+    thread_nums: usize,
+    permit_nums: usize,
+    thread_name: String,
+) -> Result<Vec<Fut::Output>>
+where
+    Fut: Future + Send + 'static,
+    Fut::Output: Send + 'static,
+{
+    // 1. build the runtime.
+    let semaphore = Semaphore::new(permit_nums);
+    let runtime = Arc::new(Runtime::with_worker_threads(
+        thread_nums,
+        Some(thread_name),
+    )?);
+
+    // 2. spawn all the tasks to the runtime with semaphore.
+    let join_handlers = runtime.try_spawn_batch(semaphore, futures).await?;
+
+    // 3. get all the result.
+    future::try_join_all(join_handlers)
+        .await
+        .map_err(|e| ErrorCode::Internal(format!("try join all futures failure, {}", e)))
+}
+```
+
+If you are interested in this Rust trick, you can read this PR: [feat: improve the parquet get splits to parallel](https://github.com/datafuselabs/databend/pull/10514).
+
+### How to Create a System Table
+
+System tables are tables that provide information about Databend's internal state, such as databases, tables, functions, and settings.
+
+If you are interested in creating system tables, you may want to check out our recently released documentation which introduces the implementation, registration, and testing of system tables using the `system.credits` table as an example.
+
+Here is a code snippet:
+
+```rust
+let table_info = TableInfo {
+    desc: "'system'.'credits'".to_string(),
+    name: "credits".to_string(),
+    ident: TableIdent::new(table_id, 0),
+    meta: TableMeta {
+        schema,
+        engine: "SystemCredits".to_string(),
+        ..Default::default()
+    },
+..Default::default()
+};
+```
+
+- [Docs | How to Create a System Table](https://databend.rs/doc/contributing/how-to-write-a-system-table)
+
+## Highlights
+
+Here are some noteworthy items recorded here, perhaps you can find something that interests you.
+
+- We suggest users to consider `unset max_storage_io_requests` to use `num_cpu` as the default value when upgrading to **1.0.17-nightly** and above.
+- Now Databend can integrate with MindsDB to provide users with machine learning workflow support. *[Bringing in-database ML to Databend](https://mindsdb.com/integrations/databend-machine-learning)*
```

Comment:
MindsDB's machine learning capability has now been integrated into Databend: [Bringing in-database ML to Databend](https://mindsdb.com/integrations/databend-machine-learning)